src/main/java/com/google/api/codegen/ApiaryConfig.java

```diff
@@ -230,4 +244,29 @@ public Field getField(Type type, String fieldName) {
   public ListMultimap<String, String> getAuthScopes() {
     return authScopes;
   }
+
+  /*
+   * @return auth type supported by the method.
+   */
+  public AuthType getAuthType(String method) {
```

Comment:
It might be conceivable that a single method does not list scopes at all though its siblings do (though I can't imagine). If we can guarantee this doesn't happen, we can set this for the entire service by looking at any method once.

Conceivably not every method lists the same scopes, though, so we would still need to disambiguate the acceptable scopes regardless.